cmake/celerity-config.cmake.in

```diff
@@ -11,18 +11,34 @@ include("${CELERITY_CMAKE_DIR}/AddToTarget.cmake")
 set(CELERITY_ORIGINAL_CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}")
 set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}" "${CELERITY_CMAKE_DIR}")
 
+if(NOT DEFINED MPI_HOME AND NOT "@MPI_HOME@x" STREQUAL x)
+  # FindMPI does not export anything akin to "MPI_DIR" that can be fed back into find_dependency.
+  set(MPI_HOME "@MPI_HOME@")
+endif()
 find_dependency(MPI 2.0 REQUIRED)
+
 find_dependency(Threads REQUIRED)
 
 if(CELERITY_SYCL_IMPL STREQUAL "hipSYCL")
-  find_dependency(hipSYCL CONFIG REQUIRED)
+  if(NOT DEFINED HIPSYCL_TARGETS AND NOT "@HIPSYCL_TARGETS@x" STREQUAL x)
+    set(HIPSYCL_TARGETS "@HIPSYCL_TARGETS@")
+  endif()
+  if(NOT DEFINED hipSYCL_DIR)
+    find_dependency(hipSYCL CONFIG REQUIRED PATHS "@hipSYCL_DIR@")
+  else()
+    find_dependency(hipSYCL CONFIG REQUIRED)
+  endif()
 elseif(CELERITY_SYCL_IMPL STREQUAL "ComputeCpp")
+  if(NOT DEFINED ComputeCpp_DIR AND NOT "@ComputeCpp_DIR@x" STREQUAL x)
+    # ComputeCpp does not support find_dependency(PATHS)
+    set(ComputeCpp_DIR "@ComputeCpp_DIR@")
+  endif()
   find_dependency(ComputeCpp REQUIRED)
 elseif(CELERITY_SYCL_IMPL STREQUAL "DPC++")
   include("${CELERITY_CMAKE_DIR}/TestDPCXX.cmake")
   test_cxx_compiler_is_dpcpp(CXX_COMPILER_IS_DPCPP)
```

Comment:
I'm just noticing some *slight* inconsistencies when it comes to naming, as on these three lines we use `DPC++`, `DPCXX` and `DPCPP` 😆

Reply:
That's why people shouldn't use special chars in product names... Well at least my `DPCXX` choice was deliberate with using `CXX` as Makefile-speak for "The DPC++ C++ compiler".